api/v1alpha1/volumereplicationgroup_types.go

```diff
@@ -18,53 +18,38 @@ package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	vrv1alpha1 "github.com/kube-storage/volume-replication-operator/api/v1alpha1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-// VolumeTakeoverControlType -- takeover control when desiredCluster is different from affinedCluster
-type VolumeTakeoverControlType string
-
-// VolumeTakeoverControlType definitions
-const (
-	// Force promote the volume in a WAN DR setting
-	ForcePromote VolumeTakeoverControlType = "ForcePromote"
-)
-
-// VolumeReplicationGroupSpec defines the desired state of VolumeReplicationGroup
+// VolumeReplicationGroup (VRG) spec declares the desired replication class
+// and replication state of all the PVCs owned by a given application, which
+// are identified via the given PVC label selector.  For every PVC created
+// by the application, the VRG operator does the following:
+// 	- Create a VolumeReplication (VR) CR to enable storage level replication
+// 	  of volume data and set the volume state to primary, secondary, etc.
+//  - Take the corresponding PV metadata in Kubernetes etcd and deposit it in
+//    the S3 store.  The url, access key and access id required to access the
+//    S3 store is specified via environment variables of the VRG operator POD,
+//    which is obtained from a secret resource.
+//  - Manage the lifecycle of VR CR and S3 data to match the CUD operations on
+//    VRG CR.
 type VolumeReplicationGroupSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Name of the application to replicate
-	ApplicationName string `json:"applicationName"`
-
-	// Application label that is used to identify its PVs
-	ApplicationLabels metav1.LabelSelector `json:"applicationLabels"`
-
-	// Cluster ID in ClusterPeers that has best storage performance affinity for the application
-	AffinedCluster string `json:"affinedCluster"`
-
-	// Desired cluster that should takeover the application from current active cluster.
-	// May be set either to:
-	// - secondary cluster ID (to migrate application or takeover application in case of a disaster)
-	// - nil (if application is not active in the affined cluster, takeback to affined cluster)
-	// +optional
-	DesiredCluster string `json:"desiredCluster,omitempty"`
-
-	// Volume Takeover Control: ForcePromote
-	// +optional
-	VolumeTakeoverControl VolumeTakeoverControlType `json:"volumeTakeoverControl,omitempty"`
+	// Label selector to identify all the PVCs that are in this group
+	// This operator will create and needs to be replicated
+	PVCLabelSelector metav1.LabelSelector `json:"pvcLabelSelector"`
```

Comment:
Rename to `PVCSelector` as label is redundant (it is of type label selector).

(note to self) I should have thought about this when making the initial comment/suggestion.

Reply:
In the spec field "ReplicationState", the prefix "Replication" is redundant too, but adds value, isn't it? :-)

Reply:
> In the spec field "ReplicationState", the prefix "Replication" is redundant too, but adds value, isn't it? :-)

True, that can be renamed to `state` as well.